operator/controllers/utils.go

```diff
@@ -482,23 +495,30 @@ func buildStatefulSet(name string, namespace string, baseImage string, isBootstr
 		log.Info("Updating StatefulSet Template Spec With ConfigVersion", STATEFULSET_V2_ANNOTATION, configVersion)
 	}
 
+	templateLabelsMap := make(map[string]string)
+	if statefulSetLabelsUpdateValue == "templateLabels" || statefulSetLabelsUpdateValue == "matchLabels" {
```

Comment:
remove `||` clause. 

Reassign with `selectorMatchLabelsMap = updateMatchLabelsForStatefulSet(name, d.Name)` if value is `templateLabels`

Reply:
More understandable code